src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -38,23 +46,181 @@
+                this.modelId = modelId;
+            }
+
+            /**
+             * @param inputTexts a {@link List} of input texts to use for inference
+             * @return this builder
+             */
+            public Builder inputTexts(List<String> inputTexts) {
+                this.inputTexts = inputTexts;
+                return this;
+            }
+
+            /**
+             * @param inputText an input text to add to the list of input texts. Repeated calls will add
+             *                  more input texts.
+             * @return this builder
+             */
+            public Builder inputText(String inputText) {
+                if (this.inputTexts != null) {
+                    this.inputTexts.add(inputText);
+                } else {
+                    this.inputTexts = new ArrayList<>();
+                    this.inputTexts.add(inputText);
+                }
+                return this;
+            }
+
+            /**
+             * @param mlAlgoParams the {@link MLAlgoParams} to use for inference.
+             * @return this builder
+             */
+            public Builder mlAlgoParams(MLAlgoParams mlAlgoParams) {
+                this.mlAlgoParams = mlAlgoParams;
+                return this;
+            }
+
+            /**
+             * @param targetResponseFilters a {@link List} of target response filters to use for
+             *                              inference
+             * @return this builder
+             */
+            public Builder targetResponseFilters(List<String> targetResponseFilters) {
+                this.targetResponseFilters = targetResponseFilters;
+                return this;
+            }
+
+            /**
+             * @param inputObjects {@link Map} of {@link String}, {@link String} on which inference needs
+             *                     to happen
+             * @return this builder
+             */
+            public Builder inputObjects(Map<String, String> inputObjects) {
+                this.inputObjects = inputObjects;
+                return this;
+            }
+
+            /**
+             * @param queryText the query text to use for similarity inference
+             * @return this builder
+             */
+            public Builder queryText(String queryText) {
+                this.queryText = queryText;
+                return this;
+            }
+
+            /**
+             * @return a new {@link InferenceRequest} object with the parameters set in this builder
+             */
+            public InferenceRequest build() {
+                return new InferenceRequest(modelId, inputTexts, mlAlgoParams, targetResponseFilters, inputObjects, queryText);
+            }
+
+        }
+    }
+
     private final MachineLearningNodeClient mlClient;
+    private final Cache<String, Boolean> modelAsymmetryCache = CacheBuilder.<String, Boolean>builder().setMaximumWeight(10_000).build();
```

Comment:
please pull the max weight number to a constant, it would be easy to find/change if needed